packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -292,9 +292,22 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
         };
       },
 
-      getWidgetRenderState({ results, helper, state, instantSearchInstance }) {
+      getWidgetRenderState({
+        results,
+        helper,
+        parent,
+        state: existingState,
+        instantSearchInstance,
+      }) {
         let isFirstPage: boolean;
         let currentPageHits: Array<Hit<THit>> = [];
+        /**
+         * We bail out of optimistic UI here, as the cache is based on search
+         * parameters, and we don't want to invalidate the cache when the search
+         * is loading.
+         */
+        const state = parent.getPreviousState() || existingState;
```

Comment:
Are there any other cases / widgets where we would want to bail out from optimistic UI? I can only think of the searchbox, since it has its own behavior linked to focus state.

Reply:
searchbox is indeed already handled, and no other widgets keep track of the previous state / results